torch/csrc/jit/passes/symbolic_shape_analysis.h

```diff
@@ -0,0 +1,22 @@
+#pragma once
+
+#include <torch/csrc/WindowsTorchApiMacro.h>
+#include <torch/csrc/jit/ir/ir.h>
+
+namespace torch {
+namespace jit {
+
+// CAUTION NOT TO BE USED, STILL A WIP, NOT STABLE
+
+TORCH_API void PropagateShapesWithShapeFunction(
+    Node* n,
+    const std::shared_ptr<Graph>& graph);
+
+TORCH_API void RegisterOperatorShapeFunction(
+    Node* n,
+    const std::shared_ptr<Graph>& graph);
+
+TORCH_API void PropagateShapesOnGraph(std::shared_ptr<Graph>& graph);
```

Comment:
If I were to use this API now (I actually am going to for the aot pipeline), am I expected to 1) set types for the graph inputs, 2) call this function?

Reply:
Yea, exactly, although it should still work reasonably well even without input types (without complete shape information obviously).